### ISSUE TITLE
Issue running the example code to test the R package

### ISSUE THREAD

**Issue by kevinrue** [beringresearch/ivis#29]
Hello,

For [JOSS review](https://github.com/openjournals/joss-reviews/issues/1596).

I am running into the following issue when running the example R code (given in this [README.md](https://github.com/beringresearch/ivis/blob/master/R-package/README.md)) in an R console in my terminal.

The package was successfully installed in an R console in my terminal as described in #28 

The main error below seems to be: `UnboundLocalError: local variable 'a' referenced before assignment`.

```
> library(ivis)
> 
> model <- ivis(k = 3, batch_size = 3)
Using TensorFlow backend.
/Users/kevin/miniconda3/envs/ivis/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: compiletime version 3.6 of module 'tensorflow.python.framework.fast_tensor_util' does not match runtime version 3.7
  return f(*args, **kwds)
> 
> X = data.matrix(iris[, 1:4])
> model = model$fit(X)
Building KNN index
/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/ivis/data/knn.py:15: FutureWarning: The default argument for metric will be removed in future version of Annoy. Please pass metric='angular' explicitly.
  index = AnnoyIndex(X.shape[1])
100%|███████████████████████████████████████████████████████████████████████| 150/150 [00:00<00:00, 127667.53it/s]
Extracting KNN from index
/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/ivis/data/knn.py:92: FutureWarning: The default argument for metric will be removed in future version of Annoy. Please pass metric='angular' explicitly.
  self.index = AnnoyIndex(n_dims)
 11%|████████▌                                                                  | 17/150 [00:00<00:00, 141.19it/s]
WARNING:tensorflow:From /Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  UnboundLocalError: local variable 'a' referenced before assignment

Detailed traceback: 
  File "/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/ivis/ivis.py", line 209, in fit
    self._fit(X, Y, shuffle_mode)
  File "/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/ivis/ivis.py", line 147, in _fit
    triplet_network(base_network(self.model_def, input_size),
  File "/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/ivis/nn/network.py", line 41, in base_network
    return default_base_network(input_shape)
  File "/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/ivis/nn/network.py", line 61, in default_base_network
    x = AlphaDropout(0.1)(x)
  File "/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/keras/engine/base_layer.py", line 457, in __call__
    output = self.call(inputs, **kwargs)
  File "/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/keras/layers/noise.py", line 165, in call
    return K
> 
> xy = model$transform(X)
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  AttributeError: 'NoneType' object has no attribute 'predict'

Detailed traceback: 
  File "/Users/kevin/miniconda3/envs/ivis/lib/python3.7/site-packages/ivis/ivis.py", line 248, in transform
    embedding = self.encoder.predict(X, verbose=self.verbose)
```

Session info reported below
```
> sessionInfo()
R version 3.6.0 (2019-04-26)
Platform: x86_64-apple-darwin18.5.0 (64-bit)
Running under: macOS Mojave 10.14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ivis_1.1.3

loaded via a namespace (and not attached):
[1] compiler_3.6.0     BiocManager_1.30.4 Matrix_1.2-17      tools_3.6.0       
[5] Rcpp_1.0.2         reticulate_1.13    grid_3.6.0         jsonlite_1.6      
[9] lattice_0.20-38  
```

**Comment by kevinrue** [comment]
Solved simultaneously to #28

**Comment by kevinrue** [comment]
I'll reopen and let you close the issue when you merge the update in a PR.

**Comment by idroz** [comment]
Addressed in #30
